### PR TITLE
DAQ module that fakes TP-creator heartbeats

### PR DESCRIPTION
**Pull request by alexbooth92**
Addition of a new plugin which generates fake tp-creator heartbeats. OpMon also implemented. The module is currently un-tested.

### CODE REVIEW

**Review by philiprodrigues — COMMENTED** [review]
This looks good. I don't see any major problems, based on looking at the code. We will have to check against `TriggerGenericMaker` to make sure we agree on the order and format of heartbeat messages, but if any changes are needed, I expect them to be minor. 

One small but annoying thing: we've abbrieviated "trigger primitive" as "TP" elsewhere in the module, so I think we should be consistent here

File: plugins/FakeTpCreatorHeartbeatMaker.cpp [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+                                                   bool const& is_first_tpset_received)
+{
+  // If it is the first TPSet received, send out a heartbeat.
+  // Else, can assume that the TPSets are already ordered by start_time. Therefore, check only
+  // that the difference between the start_time of the current TPSet and the time that the
+  // last heartbeat was sent out is greater than the specified heartbeat interval.
+  if (is_first_tpset_received)
+    return true;
+  else
+    return last_sent_heartbeat_time + m_heartbeat_interval > current_tpset_start_time;
```
Comment (philiprodrigues) [inline plugins/FakeTpCreatorHeartbeatMaker.cpp:163]: Is the ">" the right way round here? Say `last_sent_heartbeat_time` is 1000 and `m_heartbeat_interval` is 100, then we should send a heartbeat if  `current_tpset_start_time > 1100`, right?

Comment (alexbooth92) [reply plugins/FakeTpCreatorHeartbeatMaker.cpp:163]: Yes, you're right. I'll fix that.

**Comment by alexbooth92** [comment]
With regard to your comment about the abbreviation of trigger primitive, are you referring to changing Tp to TP in the module name?

**Comment by philiprodrigues** [comment]
> With regard to your comment about the abbreviation of trigger primitive, are you referring to changing Tp to TP in the module name?

Yes

**Review by philiprodrigues — APPROVED** [review]
Looks good to me. Thanks!

**Pull request merged**